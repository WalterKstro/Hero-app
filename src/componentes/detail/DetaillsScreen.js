import { Redirect, useParams } from 'react-router-dom'
import HeroCardDetail from './HeroCardDetail'
import getById from '../../switcher/getById';
const DetaillsScreen = ({history}) => {
    
    const {id_hero} = useParams()
    
    const currentHero = getById(id_hero)
    const objectBack = {'back': history.goBack, 'length' : history.length, 'push': history.push}
    
    if(!currentHero) {
        return <Redirect to="/"/>
    }

    return (
        <>
        {
            <HeroCardDetail
                {...currentHero}
                {...objectBack}
            />
        }
        </>
    )
}

export default DetaillsScreen
