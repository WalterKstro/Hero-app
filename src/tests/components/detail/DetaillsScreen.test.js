import { mount } from "enzyme"
import { MemoryRouter, Route } from "react-router"
import DetaillsScreen from "../../../componentes/detail/DetaillsScreen"
import Error404Page from "../../../componentes/error404/Error404Page"

describe('Testing in componente <DetaillsScreen/>', () => {
    
    const history = {
        goBack : jest.fn(),
        push : jest.fn(),
        length : 1 
    }


    test('should display the component <Redirect/> and redirect to home, if not found an heroe', () => {
        const rendered = mount(
            <MemoryRouter initialEntries={['/heroe']}>
                <DetaillsScreen history={history}/>
            </MemoryRouter>
        )
        expect(rendered).toMatchSnapshot()
        expect(rendered.find('Redirect').exists()).toBe(true)
    })
    
    test('should display the component <Error404Page/> if the URL is invalid', () => {
        const urlInvalid = 'asdf456we'
        const rendered = mount(
            <MemoryRouter initialEntries={[`${urlInvalid}`]}>
                <Route component={ Error404Page }/>
            </MemoryRouter>
        )
        
        expect(rendered).toMatchSnapshot()
        expect(rendered.find('h1').exists()).toBe(true)
        expect(rendered.find('h1').text()).toBe('Error 404 Page not Found')
    })


    test('should found an heroe if exist param and it is found a heroe in the consult', () => {
        const id_heroe = 'dc-batman'

        const rendered = mount(
            <MemoryRouter initialEntries={[`/hero/${id_heroe}`]}>
                <Route path="/hero/:id_hero" component={DetaillsScreen}/>
            </MemoryRouter>
        )
        expect(rendered.find('.btn-outline-primary').exists()).toBe(true)
    })
    
    
})
