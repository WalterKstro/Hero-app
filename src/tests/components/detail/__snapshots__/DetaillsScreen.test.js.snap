// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in componente <DetaillsScreen/> should display the component <Error404Page/> if the URL is invalid 1`] = `
<div>
  <h1>
    Error 404 Page not Found
  </h1>
</div>
`;

exports[`Testing in componente <DetaillsScreen/> should display the component <Redirect/> and redirect to home, if not found an heroe 1`] = `""`;
